### PR TITLE
Fix counter output with RTP & file

### PR DESCRIPTION
**Pull request by lars18th**
Fix incorrect output calculations with file output and RTP enabled.

### CODE REVIEW

**Comment by freddy36** [comment]
Can you explain what exactly is the problem?
I would expect that written2 (to file) is always 12 bytes less than written (to UDP/network) under normal conditions so your new if condition should never trigger.
We intentionally did not add the RTP header size to the o->traffic/o->traffic_period counters as we assumed that these are used for bitrate caculation. And there I would expect it to be the MPEG-TS bitrate (not including RTP overhead).

**Comment by lars18th** [comment]
Hi @freddy36 ,

> Can you explain what exactly is the problem?

Yes. This code is similar to the one added to `ts_frame_write()` in this commit: https://github.com/gfto/mptsd/commit/6a59e479cf5ae5cb3e2c0221e61a7c5343553858

In this commit the source change for the general UDP output, focus on the line that writes to the output file... that requires to be before the counter increase. The reason is because this: when the network output is _disabled_ the counter don't raises. In this particular case the counter requires to use the data writed to the file, beacuse the network writen data is 0.

However, in this particular commit your code for the RTP output is not included. And your PR of the RTP output doesn't includes this one. Therefore, your code is based on the "old" code... that writes to the file _after_ the counter changes. The solution then is to mimic the same structure. And that is the goal of this PR.

In particular to your question, if you check the code then you'll see that if both: RTP and file data are writed, then the `written` value will be greater than `written2` (the data writed to the file). This will be allways true because the RTP header. However, if the file is enabled and the network is off, but RTP is selected (a very strange configuration, not noticed before because it doesn't have sense) then in this particular (and unique) use case, the `written2` value will be: TS data length without any RTP header. And `written` be zero (logically). Then to not complicate the code that increases the counter with the data _less the RTP header_ will continue there. And therefore, the value of `written` will be increased. But only in case of network==0. This seems not very efficient, but in fact the use case is extremelly rare. So don't care about it.

In other words:
```
	if (written2 > written)
		written = written2 + RTP_HEADER_SIZE;
```
- Every time `written2 < written`, and only if `written == 0` then `written = written2 + RTP_HEADER_SIZE` that is equivalent to the regular `written` to the network that includes the RTP header.


Belive me that I've checked the code executing this particular case. And the new code works as expected. The  o->traffic/o->traffic_period counters will not have any RTP header size overhead in any case.

Regards.

**Comment by freddy36** [comment]
Ok, for that very special use case it makes sense. Please merge.
